HW04.Rmd

```diff
@@ -1,126 +1,215 @@
-
-```{r intent}
-
-```
-
-## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
-
-```{r age_&_sex}
-
-```
-
-
-## How many white males with at least a high school education were killed by guns in 2012?
-
-```{r education}
-
-```
-
-
-## Which season of the year has the most gun deaths?
-
-Assume that:
-
-* Winter = January-March
-* Spring = April-June
-* Summer = July-September
-* Fall = October-December
-
-```{r season}
-
-```
-
-
-# These are more open ended questions from here on out, you could look at making multiple summaries or types of plots. 
-
-### Are whites who are killed by guns more likely to die because of suicide or homicide? How does this compare to blacks and hispanics?
-
-```{r race}
-
-```
-
-(hint maybe looking at percentages would be good)
-```{r race_2}
-
-```
-
-
-### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
-
-```{r police}
-
-```
-
-Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
-
-```{r stats?}
-
-
-```
-
-
-```{r police_2}
-
-```
-
-```{r stats_2?}
-
-
-```
-
-
-Finally, all together now:
-```{r police_3}
-
-```
-
-
-## Session info
-
-```{r}
-# always good to have this for reproducibility purposes
-devtools::session_info()
-```
-
+---
+title: "Exploring Gun Deaths in America"
+output: github_document
+---
+
+## Objectives
+* To explore gun death data curated by FiveThirtyEight [original data available here](https://github.com/fivethirtyeight/guns-data) by
+    * Month
+    * Intent
+    * Age and sex
+    * Education
+    * Season
+    * Intent by race
+    * Police-relatedness 
+
+## Read in the data
+
+The gun_deaths.csv is provided in the repository. 
+```{r load-data, message=FALSE, warning=FALSE}
+library(tidyverse)    # load tidyverse packages, including ggplot2
+library(readr)
+library(knitr)
+library(usethis)
+
+# read in the data
+gun_deaths <- read_csv("gun_deaths.csv")
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+Knitr is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+death_month <- gun_deaths %>% 
+  group_by(month) %>% 
+  summarise('Number of Gun Deaths' = n())
+kable(death_month)
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+p1 <- ggplot(death_month, aes(x = month.abb[month], y = `Number of Gun Deaths`)) +
```

Comment:
Wow, I didn't know there is a month.abb variable in R! This is a very useful built-in variable.

Reply:
That's indeed something new I just learned too!

Reply:
I just realized by looking at the mardown that the axis month labels are disorgnized somehow. It starts with april and ends with september. Is there a way to fix this and still use month.abb?